base/challenge-skeleton/README.md

```diff
@@ -0,0 +1,40 @@
+# Quickstart guide to writing a challenge:
+
+* Put your challenge binary under challenge/image/ and update challenge/image/Dockerfile.
+* By default /chroot/home/user/chal will be run. The command line can be changed in challenge/runtime-files/nsjail.cfg.
+* Try the challenge locally with `make test-docker`.
+* If you have a cluster ready, try it remotely with `make start && make port-forward`.
+
+## healthcheck
+
+* The healthcheck needs to run a webserver on port 8080 that responds to /healthz requests.
+ * 200 if the challenge is healthy, otherwise an error (e.g. 400).
+* The base image comes preloaded with pwntools.
+* If your exploit is written in python, just drop it in healthcheck/exploit/doit.py.
+* Note: for defense in depth, the exploit is encrypted before being added to the docker image.
+
+# Directory layout:
+
+## /
+
+* chal.conf: Configure if the challenge should be deployed to the cluster if it's publicly accessible and if it has a healthcheck.
+* Makefile: Anything that needs to be run before deploying a challenge.
+* challenge: The files for the challenge container.
+* healthcheck: The files for the healthcheck container (optional).
+* advanced: The kubernetes config used to deploy the challenge. E.g. add a tmpfs here or change the port.
+* .gen: A directory for files generated by make.
+
+## /challenge
+
+* image/: Dockerfile and files of the challenge image. Note that for defense in depth, you shouldn't put secrets in here.
+* runtime-files: Files to be added to the challenge at runtime. Secrets are ok, but note that there's a 1MB size limit for the whole directory. Supports live updates without a challenge restart.
+* runtime-files/flag: The flag (duh).
+* runtime-files/nsjail.cfg: The configuration for the nsjail sandbox. Configure here limits and what you want to expose to the challenge chroot.
+* runtime-files/pow: Difficulty of the proof-of-work. 0 means disabled.
+
+## /healthcheck
+
+* /: The whole directory is optional. If you're not going to implement a healthcheck, feel free to delete this.
+* image/: Dockerfile and files of the healthcheck.
+* exploit/: The exploit code to check if the challenge is healthy. This directory will be encrypted for defense in depth.
```

Comment:
We should explain where it will be mounted.

Reply:
re this and the other comments: all of this is outdated and was describing the "we need to use secrets" approach